src/webview/webview2/mod.rs

```diff
@@ -299,6 +299,49 @@ window.addEventListener('mousemove', (e) => window.chrome.webview.postMessage('_
     }
     .map_err(webview2_com::Error::WindowsError)?;
 
+    if let Some(nav_callback) = attributes.navigation_handler {
+      unsafe {
+        let nav_starting_callback = nav_callback.clone();
+        webview
+          .NavigationStarting(
+            NavigationStartingEventHandler::create(Box::new(move |_, args| {
+              if let Some(args) = args {
+                let mut uri = PWSTR::default();
+                args.Uri(&mut uri)?;
+                let uri = take_pwstr(uri);
+                
+                let cancel = nav_starting_callback(uri, false);
+                
+                args.SetCancel(cancel)?;
+              }
+  
+              Ok(())
+            })),
+            &mut token
+          )
+          .map_err(webview2_com::Error::WindowsError)?;
+        let new_window_callback = nav_callback.clone();
+        webview
+          .NewWindowRequested(
+            NewWindowRequestedEventHandler::create(Box::new(move |_, args| {
+              if let Some(args) = args {
+                let mut uri = PWSTR::default();
+                args.Uri(&mut uri)?;
+                let uri = take_pwstr(uri);
+                
+                let cancel = new_window_callback(uri, false);
+                
+                args.SetHandled(cancel)?;
+              }
+              
+              Ok(())
+            })),
+            &mut token
+          )
+          .map_err(webview2_com::Error::WindowsError)?;
+      }
+    }
+
```

Comment:
This should be in another PR because it is not the same thing.

Reply:
That's fair, even though I can see users wanting pretty similar behaviour from both. I'll open a PR based on these changes that basically adds the above but using a separate `new_window` handler